service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerMidEventTest.java

```diff
@@ -287,23 +298,98 @@ void shouldNotReturnAnErrorWhenAllChildrenHaveAFinalOrderIssuedButIssuingANonFin
                 createChild("Barry", true));
 
             AboutToStartOrSubmitCallbackResponse response = postMidEvent(
-                buildCaseDetails(null, chuckleBrothers, BLANK_ORDER, null), "add-final-order-flags");
+                buildCaseDetails(null, chuckleBrothers, BLANK_ORDER, null), callbackType);
 
             assertThat(response.getErrors()).isNull();
         }
     }
 
+    @Nested
+    class DischargeCaseOrderPreparation {
+
+        private final String callbackType = "prepare-selected-order";
+
+        @Test
+        void shouldReturnErrorWhenCurrentOrderIsDischargeOfCareOrderAndNoExistingCareOrders() {
+            GeneratedOrder order = order("Supervision order", "1 May 2020");
+            CaseDetails caseDetails = caseWithOrders(DISCHARGE_OF_CARE_ORDER, order);
+
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, callbackType);
+
+            assertThat(callbackResponse.getData().get("singleCareOrder_label")).isNull();
+            assertThat(callbackResponse.getData().get("multipleCareOrder_label")).isNull();
+            assertThat(callbackResponse.getData().get("careOrderSelector")).isNull();
+            assertThat(callbackResponse.getErrors()).containsExactly("No care orders to be discharged");
+        }
+
+        @Test
+        void shouldPopulateLabelForSingleCareOrder() {
+            GeneratedOrder order = order("Interim care order", "1 May 2020", child("John", "Smith"));
+
+            CaseDetails caseDetails = caseWithOrders(DISCHARGE_OF_CARE_ORDER, order);
+
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, callbackType);
+
+            assertThat(callbackResponse.getData().get("singleCareOrder_label"))
+                .isEqualTo("Create discharge of care order for John Smith");
+            assertThat(callbackResponse.getData().get("multipleCareOrder_label")).isNull();
+            assertThat(callbackResponse.getData().get("careOrderSelector")).isNull();
+            assertThat(callbackResponse.getErrors()).isEmpty();
+        }
+
+        @Test
+        void shouldPopulateLabelAndSelectorForMultipleCareOrders() {
+            GeneratedOrder order1 = order("Interim care order", "1 June 2019", child("John", "Smith"));
+            GeneratedOrder order2 = order("Final care order", "12 June 2019",
+                child("John", "Smith"),
+                child("Alex", "White"));
+
+            CaseDetails caseDetails = caseWithOrders(DISCHARGE_OF_CARE_ORDER, order1, order2);
+
+            AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, callbackType);
+
+            CaseData updatedCaseData = extractCaseData(callbackResponse);
+
+            assertThat(callbackResponse.getData().get("multipleCareOrder_label"))
+                .isEqualTo("Order 1: John Smith, 1 June 2019\nOrder 2: John Smith and Alex White, 12 June 2019");
```

Comment:
Interesting you used lineSeparator in implementation but not here. I guess this will fail on people with Windows machines - the code will generate \n\r and the code will expect \n.

Reply:
fixed

Reply:
Not fixed? It's still \n vs lineSeparator? 🤔